### PR TITLE
Protobufs+UDP

### PR DESCRIPTION
**Pull request by AllanYangZhou**

### CODE REVIEW

File: dawn/main/ansible/ansible.proto [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    repeated Gamepad gamepad = 1;
+}
+
+message ConsoleOutput {
+    string output = 1;
+}
+
+message AnsibleMessage {
+    string type = 1;
+    bytes data = 2;
```
Comment (nikitakit) [inline dawn/main/ansible/ansible.proto:23]: I notice you're storing bytes here, and then parsing them as proto again. Have you tried using proto3's "Any" type instead? (I'll need to check, but I expect the "Any" type to be more space-efficient while also offering a better API)

Comment (nikitakit) [inline dawn/main/ansible/ansible.proto:23]: Actually nevermind, Any is the same as what you're doing here while being extremely space-inefficient. I think OneOf (a union) is what you actually want.

Comment (AllanYangZhou) [inline dawn/main/ansible/ansible.proto:23]: Thanks, I'll look into OneOf, looks like what I want.

**Comment by AllanYangZhou** [comment]
Outdated, look at #282 for ansible work instead.

**Pull request closed without merging**